components/camel-jaxb/src/main/java/org/apache/camel/converter/jaxb/JaxbDataFormat.java

```diff
@@ -206,9 +206,10 @@ void doMarshal(Exchange exchange, Object graph, OutputStream stream, Marshaller
         // only marshal if its possible
         if (introspector.isElement(element)) {
             if (asXmlStreamWriter(exchange)) {
-                XMLStreamWriter writer = typeConverter.convertTo(XMLStreamWriter.class, stream);
+                XMLStreamWriter writer = typeConverter.convertTo(XMLStreamWriter.class, exchange, stream);
                 if (needFiltering(exchange)) {
-                    writer = new FilteringXmlStreamWriter(writer);
+                    String charset = exchange.getProperty(Exchange.CHARSET_NAME, String.class);
```

Comment:
Could you please change the method signature to `doMarshal(..., OutputStream stream, String charset, ...)` and get the charset from there ?